File: dotnet/src/MyDotey.SCF.Simple/Source/StringProperty/PropertieFile/PropertiesFileConfigurationSource.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

using NLog;
using Kajabity.Tools.Java;

namespace MyDotey.SCF.Source.StringProperty.PropertiesFile
{
    /**
     * @author koqizhao
     *
     * May 17, 2018
     */
    public class PropertiesFileConfigurationSource : StringPropertyConfigurationSource<PropertiesFileConfigurationSourceConfig>
    {
        private static Logger LOGGER = LogManager.GetCurrentClassLogger(typeof(PropertiesFileConfigurationSource));

        private JavaProperties _properties;

        public PropertiesFileConfigurationSource(PropertiesFileConfigurationSourceConfig config)
            : base(config)
        {
            _properties = new JavaProperties();

            try
            {
                using (Stream @is = new FileStream(config.getFileName(), FileMode.Open))
                {
                    if (@is == null)
                        LOGGER.Warn("file not found: {0}", config.getFileName());

                    _properties.Load(@is, new UTF8Encoding(false));
                }
            }
            catch (Exception e)
            {
                LOGGER.Warn(e, "failed to load properties file: " + config.getFileName());
            }
        }

        public override String getPropertyValue(String key)
        {
            if (_properties == null)
                return null;

            return _properties.GetProperty(key);
        }
    }
}

File: java/scf-simple/src/main/java/org/mydotey/scf/source/stringproperty/propertiesfile/PropertiesFileConfigurationSource.java
package org.mydotey.scf.source.stringproperty.propertiesfile;

import java.io.InputStream;
import java.util.Properties;

import org.mydotey.scf.source.stringproperty.StringPropertyConfigurationSource;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * @author koqizhao
 *
 * May 17, 2018
 */
public class PropertiesFileConfigurationSource
        extends StringPropertyConfigurationSource<PropertiesFileConfigurationSourceConfig> {

    private static final Logger LOGGER = LoggerFactory.getLogger(PropertiesFileConfigurationSource.class);

    private Properties _properties;

    public PropertiesFileConfigurationSource(PropertiesFileConfigurationSourceConfig config) {
        super(config);

        _properties = new Properties();
        try (InputStream is = Thread.currentThread().getContextClassLoader()
                .getResourceAsStream(config.getFileName())) {
            if (is == null)
                LOGGER.warn("file not found: {}", config.getFileName());

            _properties.load(is);
        } catch (Exception e) {
            LOGGER.warn("failed to load properties file: " + config.getFileName(), e);
        }
    }

    @Override
    public String getPropertyValue(String key) {
        if (_properties == null)
            return null;

        return _properties.getProperty(key);
    }

}
